kuwala/core/canvas/src/data/state/models/CanvasModel.js

```diff
@@ -82,26 +85,48 @@ const CanvasModel = {
         const { source, target } = params;
         const targetElement = getElementById(elements, target);
 
-        if (targetElement && targetElement.type === TRANSFORMATION_BLOCK ) {
-            if (
-                targetElement.data.transformationBlock.connectedSourceNodeIds.length < targetElement.data.transformationCatalogItem.maxNumberOfInputBlocks &&
-                !connectionExists
+        if (
+            targetElement &&
+            targetElement.type === TRANSFORMATION_BLOCK &&
+            targetElement.data.transformationBlock.connectedSourceNodeIds.length < targetElement.data.transformationCatalogItem.maxNumberOfInputBlocks &&
+            !connectionExists
+        ) {
+            addConnectedEdgeToBlock({
+                source,
+                target,
+                updateDataBlock,
+                updateTransformationBlock,
+                updateExportBlock
+            });
+            addElement(edgeToAdd);
+        } else if (
+            targetElement &&
+            targetElement.type === EXPORT_BLOCK &&
+            targetElement.data.exportBlock.connectedSourceNodeIds.length < targetElement.data.exportCatalogItem.maxNumberOfInputBlocks &&
+            !connectionExists
+        ) {
+            addConnectedEdgeToBlock({
+                source,
+                target,
+                updateDataBlock,
+                updateTransformationBlock,
+                updateExportBlock
+            });
+            addElement(edgeToAdd);
+        } else {
```

Comment:
Please combine the two if statements into one since the functions inside the blocks are identical.